sklearn/preprocessing/_data.py

```diff
@@ -1728,7 +1728,7 @@ def normalize(X, norm='l2', axis=1, copy=True, return_norm=False):
         elif norm == 'l2':
             norms = row_norms(X)
         elif norm == 'max':
-            norms = np.max(X, axis=1)
+            norms = np.max(np.abs(X), axis=1)
```

Comment:
Should we instead avoid a copy with something like `np.maximum(abs(np.min(X, axis=1)), np.max(X, axis=1))`? Or is that too crazy?

Reply:
As far as I can tell, for vectors, max norm and infinity norm are equal and correspond to taking the maximum of the absolute values: https://en.wikipedia.org/wiki/Norm_(mathematics)#Maximum_norm_(special_case_of:_infinity_norm,_uniform_norm,_or_supremum_norm)

Reply:
My point here is that `abs(X)` copies X and transforms it. For large data, `np.maximum(abs(np.min(X, axis=1)), np.max(X, axis=1))` would be more memory efficient but give the same result.

Reply:
I agree with you that avoiding a copy is worth here, I was just answering to your question on norms on top of this one.

Reply:
But for now we have kept it copying?

Reply:
Sorry, my bad. Should be good now.